state-chain/pallets/cf-swapping/src/tests/dca.rs

```diff
@@ -2,6 +2,102 @@ use super::*;
 
 const CHUNK_INTERVAL: u32 = 3;
 
+#[track_caller]
+fn setup_dca_swap(
+	number_of_chunks: u32,
+	chunk_interval: u32,
+	refund_params: Option<TestRefundParams>,
+) {
+	// Sanity check that the test started at the correct block
+	assert_eq!(System::block_number(), INIT_BLOCK);
+
+	// Start the dca swap
+	insert_swaps(&[TestSwapParams::new(
+		Some(DcaParameters { number_of_chunks, chunk_interval }),
+		refund_params,
+		false,
+	)]);
+
+	// Check that the swap request was received;
+	assert_has_matching_event!(
+		Test,
+		RuntimeEvent::Swapping(Event::SwapRequested {
+			swap_request_id: SWAP_REQUEST_ID,
+			input_amount: INPUT_AMOUNT,
+			dca_parameters,
+			..
+		}) if dca_parameters == &Some(DcaParameters { number_of_chunks, chunk_interval })
+	);
+
+	// Check that the first chunk was scheduled
+	let chunk_amount = INPUT_AMOUNT / number_of_chunks as u128;
+	assert_has_matching_event!(
+		Test,
+		RuntimeEvent::Swapping(Event::SwapScheduled {
+			swap_request_id: SWAP_REQUEST_ID,
+			swap_id: 1,
+			input_amount,
+			execute_at,
+			..
+		}) if *input_amount == chunk_amount
+			&& *execute_at == INIT_BLOCK + SWAP_DELAY_BLOCKS as u64
+	);
+
+	// Check the DCA state is correct
+	assert_eq!(
+		get_dca_state(SWAP_REQUEST_ID),
+		DcaState {
+			status: DcaStatus::ChunkScheduled(1),
+			remaining_input_amount: INPUT_AMOUNT - chunk_amount,
+			remaining_chunks: number_of_chunks - 1,
+			chunk_interval,
+			accumulated_output_amount: 0,
+		}
+	);
+}
+
+#[track_caller]
+fn assert_chunk_1_executed(
```

Comment:
This function is always called with the same parameters. I know that some of these may be useful to change in the future for other tests, but I also feel like the tests would be more clear if we removed either some or all of them (e.g. by highlighting that this isn't the focus of the tests). No strong opinion though.

I do think it is a little strage that we have to pass both `chunk_amount_after_fee` and `chunk_amount`. Once you have one you should be able to derive the other (using `BROKER_FEE_BPS`). As another option, you can assume `CHUNK_AMOUNT` and `CHUNK_AMOUNT_AFTER_FEE` are used in all tests and remove the amount parameters.